theano/gof/toolbox.py

```diff
@@ -100,8 +103,15 @@ def __init__(self, old_r, new_r, old_r_val, new_r_val, reason,
         self.reason = reason
         self.old_graph = old_graph
         self.new_graph = new_graph
+        self.full_err = None
+        if isinstance(old_r, str):
+            assert (new_r is None and old_r_val is None and new_r_val is None and
+                    reason is None and old_graph is None and new_graph is None)
+            self.full_err = old_r
```

Comment:
I'm not opposed to the change in the code, I just think that there should be a comment in the code of `__init__` and `__str__` explaining how the different paths taken relate to these two modes.
If you think this is already clear enough, I won't block the PR, and blame my cold for why it took me a while to understand.